src/app/@ansyn/ansyn/modules/menu-items/tools/export-maps-popup/export-maps-popup.component.ts

```diff
@@ -1,99 +1,304 @@
 	}
 
-	exportMapsToPng() {
-			toBlob(document.querySelector(this.config.target), {
-				filter: (element) => {
-					if (element.tagName === 'CANVAS') {
-						return element.width > 0;
-					}
-					return !(element.classList && this.config.excludeClasses.some( excludeClass => element.classList.contains(excludeClass)));
+	ngOnInit(): void {
+	}
+
+	ngOnDestroy(): void {
+	}
+
+	advancedExportMaps(exportMetadata: IExportMapMetadata) {
+		const mapToBeExport = this.imageryCommunicatorService.provide(this.pdfExportMapId);
+
+		if (!mapToBeExport) {
+			throw new Error('No Such Map Error');
+		}
+
+
+		mapToBeExport.exportMap(exportMetadata).pipe(
+			mergeMap( (exportMapData) => {
+				if (exportMetadata.extra.north) {
+					return new Observable( obs => {
+						const rotation = mapToBeExport.getRotation();
+						const northImage = new Image();
+						northImage.onload = function() {
+							const { width, height } = northImage;
+							northImage.style.transform = `rotate(${rotation}rad)`;
+							const canvas = document.createElement('canvas');
+							canvas.width = width;
+							canvas.height = height;
+							const x = canvas.width / 2, y = canvas.height / 2;
+							const ctx = canvas.getContext('2d');
+							ctx.translate(x, y);
+							ctx.rotate(rotation);
+							ctx.drawImage(northImage, -width / 2, -height / 2, width, height);
+							ctx.rotate(-rotation);
+							ctx.translate(-x, -y);
+							obs.next({...exportMapData, compass: canvas});
+							obs.complete();
+						};
+						northImage.src = 'assets/icons/full-north.svg';
+					})
+				}
+				return of(exportMapData);
+			}),
+			tap( (exportMapData: IExportMapData) => {
+				const {size, extra} = exportMetadata;
+				const doc = new jsPDF('landscape', undefined, this.pageSize);
+				doc.addImage(exportMapData.canvas.toDataURL('image/jpeg'), 'JPEG', 0, 0, exportMetadata.size[0], exportMetadata.size[1]);
+				if (extra.descriptions) {
+					doc.rect(0, 0, size[0], 5, 'F');
+					doc.setTextColor(255, 255, 255);
+					doc.setFontSize(11);
+					const loadOverlay = mapToBeExport.mapSettings.data.overlay;
+					const desc = Boolean(loadOverlay) ? this.getDescriptionFromOverlay(loadOverlay) : 'Base Map';
+					doc.text(desc, size[0] / 2, 5, {align: 'center', baseline: 'bottom'});
+				}
+				if (exportMapData.compass) {
+					doc.addImage(exportMapData.compass.toDataURL('image/png'), 'PNG', 0, 0, 25, 25); // we use png for transparent compass
 				}
-			}).then(blob => {
-				this.store$.dispatch(new SetMinimalistViewModeAction(false));
-				this.isDownloadAvailable = true;
-				this.mapBlob.emit(blob)
-			}).catch( err => {
+
+				doc.save('map.pdf');
+				this.logger.info(LOGS.success);
+			}),
+			catchError( (err) => {
+				console.error(err);
 				this.logger.error(LOGS.failed);
-				this.closeModal();
+				this.store$.dispatch(new SetToastMessageAction({toastText: `can't export map,use basic export instead`}));
```

Comment:
add the text 'can't export map,use basic export instead' to the toastMessages object